### ISSUE TITLE
MuSR: 3He Refrigerator Testing

### ISSUE THREAD

**Issue by JamesKingWork** [ISISComputingGroup/IBEX#6243]
As a MuSR scientist, I would like the 3He Refrigerator IOC to be sufficiently tested against hardware.

Acceptance criteria:

- [ ] Communications to the device is tested against hardware
- [ ] Control logic of the IOC is tested against hardware

Notes:

- The 3He refrigerator is a Heliox/ITC503 based system
- LabVIEW VIs are at C:\LabVIEW Modules\Oxford Software\Muon Heliox

3He Refrigerator tickets:

**Comment by JamesKingWork** [comment]
They do want to use this for the upcoming cycle and cryogenics have confirmed that they can set up the machine shortly after the software is completed

**Comment by JamesKingWork** [comment]
Impeded waiting on https://github.com/ISISComputingGroup/IBEX/issues/6294 and https://github.com/ISISComputingGroup/IBEX/issues/6240. We could do an initial test after the communications are reworked and then test the autocondense separately

**Comment by JamesKingWork** [comment]
The testing for this device is currently being postponed to enable EMU mercury itc and MUSR soak testing

**Comment by JamesKingWork** [comment]
The current plan is to move back to SECI if this device is needed in the next cycle. Removing it from the board and impeded, will propose again when MUSR want to test

**Comment by rerpha** [comment]
so far we have been working on 
OPI https://github.com/ISISComputingGroup/ibex_gui/tree/Ticket6243_hardware_testing 
support https://github.com/ISISComputingGroup/EPICS-hlx503/tree/field_test_fixes

**Comment by rerpha** [comment]
we may need to use https://github.com/ISISComputingGroup/IBEX/issues/7143 if possible as the state machine seems to be sending commands in sequence too quickly!

**Comment by rerpha** [comment]
im going to be a bit cheeky and merge the above branches into master so we can have a built copy in case my copied-over version does not work in the morning. hopefully the build completes!

**Comment by rerpha** [comment]
in a nutshell, everything seemed to work and we managed a recondense as well as skipping through the first part of a recondense. setting setpoints for sorb/he3 pot now work well. I will be setting up on WISH today so will put this in review once they are happy

**Comment by rerpha** [comment]
repointing - some other things are needed: 

 - It would be better if the threshold for switching between the HE3POTLO and HE3POTHI thermometers was separate to the threshold used for switching between high and low temperature mode. I think we discussed this when we implemented the thermometer switching, but ended up using the same threshold for speed. Unfortunately this didn’t work out during real usage.
 - There are a lot of communications drop outs for some reason which causes the reported temperature to drop down to zero. See attached.
 - The switching the heater off every time a new setpoint is sent is messing up the PIDs, causing very noticeable temperature drops and hence time delays each time the setpoint is changed (again, see graph).

**Comment by rerpha** [comment]
actually, i will create a new ticket for this, as the hardware testing has now happened and the other things can be considered separate tasks. 

Reviewing this one won't really require much reviewing as WISH used the sorb but would like some tweaks made. I might just close it, but feel free to open it again and review if you disagree

**Comment by rerpha** [comment]
https://github.com/ISISComputingGroup/IBEX/issues/7168 is the tweaks ticket
